TextStyling-Kotlin/app/src/main/java/com/android/example/text/styling/parser/Parser.kt

```diff
@@ -48,21 +49,20 @@ object Parser {
         val matcher = patternQuote.matcher(string)
         var lastStartIndex = 0
 
-        while (matcher.find(lastStartIndex)) {
-            val startIndex = matcher.start()
-            val endIndex = matcher.end()
-            // we found a quote
-            if (lastStartIndex < startIndex) {
-                // check what was before the quote
-                val text = string.subSequence(lastStartIndex, startIndex)
-                parents.addAll(findElements(text, pattern))
+        generateSequence { matcher.findBoundaries(lastStartIndex) }
```

Comment:
add a comment explaining what `generateSequence` does. For devs new to Kotlin this is harder to understand than the `while`. (same for the other generateSequence)
Extra Q: is there any performance downside of this compared to while?

Reply:
The sequence basically relies on an iterator, and this iterator ends when the returned value is null. So it could be said it's a little less performant (it needs to create an iterator), but that's all.

Then it uses a while too to go through the iterator. Here's the generated bytecode:

![image](https://user-images.githubusercontent.com/2178243/36087664-d317246a-0fd3-11e8-85df-89537b87aa60.png)

As you see, in the end the code is pretty similar to the original one.